### PR TITLE
Fix: Upgrade gopkg.in/yaml.v2 to gopkg.in/yaml.v3 due to security issue

### PR DESCRIPTION
**Pull request by rafaeljusto**
From the security report:
> An issue in the Unmarshal function in Go-Yaml v3 causes the program to crash
> when attempting to deserialize invalid input.

While upgrading to version 3, there were some required changes:
* Force the encoder to use 2 spaces for indentation
* Rewrite tests so lists are indented

References:
https://nvd.nist.gov/vuln/detail/CVE-2022-28948
https://github.com/go-yaml/yaml/issues/666
https://github.com/advisories/GHSA-hp87-p4gw-j4gq

### CODE REVIEW

**Comment by rafaeljusto** [comment]
The main alert came from here:
https://github.com/advisories/GHSA-hp87-p4gw-j4gq

**Comment by rafaeljusto** [comment]
🤔 Looks like the output indentation changed from 2 to 4 spaces in the new version.

**Review by shane-tw — APPROVED** [review]

**Comment by ripexz** [comment]
Am I misreading the linked issues/reports or was the vulnerability *only* present in v3 and not v2 (which we're currently using)? Granted it's now fixed in v3 as well, but arguably this wasn't necessary unless we intend to use some v3 features 🤔

**Comment by rafaeljusto** [comment]
Yep, you read it right 😄 , the biggest pain is that the [Github Alert](https://github.com/advisories/GHSA-hp87-p4gw-j4gq) (for some strange reason) decided to annoy about anything before version 3.0.0:

image

So to get rid of the repository warning, we actually need to upgrade to version 3:
image

**Comment by ripexz** [comment]
They just released `3.0.1`, might as well go for that then 😅

**Comment by rafaeljusto** [comment]
Thanks @ripexz ! Upgraded there now 🙂

File: go.mod [hunk 1]
```diff
@@ -8,6 +8,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/teamwork/test v0.0.0-20181126061546-2ff8918eb6a4
 	github.com/teamwork/utils v0.0.0-20190114034940-d6a1f27ce92c
-	gopkg.in/yaml.v2 v2.2.2
+	gopkg.in/yaml.v2 v2.4.0 // indirect
```
Comment (ripexz) [inline go.mod:11]: What's causing the indirect dep here, won't we get security alerts on the repo anyway? 🤔

Comment (rafaeljusto) [reply go.mod:11]: Nice catch 👍  it's coming from the outdated `github.com/imdario/mergo` dependency, I will update it there.

**Review by ripexz — APPROVED** [review]

**Pull request merged**